sensu_plugin/utils.py

```diff
@@ -68,3 +68,63 @@ def deep_merge(dict_one, dict_two):
         else:
             merged[key] = value
     return merged
+
+
+def map_v2_event_into_v1(event):
+    '''
+    Helper method to convert Sensu 2.x event into Sensu 1.x event.
+    '''
+
+    # return the event if it has already been mapped
+    if "v2_event_mapped_into_v1" in event:
+        return event
+
+    # convert the event into a dict
+    event = json.loads(event)
+
+    # Trigger mapping code if enity exists and client does not
+    if "client" not in event and "entity" in event:
+        event['client'] = event['entity']
+
+        # Fill in missing client attributes
+        if "name" not in event['client']:
+            event['client']['name'] = event['entity']['id']
+
+        if "subscribers" not in event['client']:
+            event['client']['subscribers'] = event['entity']['subscriptions']
+
+        # Fill in renamed check attributes expected in 1.4 event
+        if "subscribers" not in event['check']:
+            event['check']['subscribers'] = event['check']['subscriptions']
+
+        if "source" not in event['check']:
+            event['check']['source'] = event['check']['proxy_entity_id']
+
+        # Mimic 1.4 event action based on 2.0 event state
+        #  action used in logs and fluentd plugins handlers
+        action_state_mapping = {'flapping': 'flapping', 'passing': 'resolve',
+                                'failing': 'create'}
+
+        if "state" in event['check']:
+            state = event['check']['state']
+        else:
+            state = "unknown::2.0_event"
+
+        if "action" not in event:
+            event['action'] = action_state_mapping[state.lower()]
```

Comment:
in the ruby i have a fallback to use save state variable as event['action']  if state.lower() is not a valid key in the action_state_mapping. the python change is something like this

if "action" not in event:
  if state.lower() in action_state_mapping:
    event['action'] = action_state_mapping[state.lower()]
  else:
   event['action'] = state